REVIEWING.md

```diff
@@ -0,0 +1,125 @@
+    * If not and the change is large: back up and potentially back-port a spec in the original issue.
+
+### Software interfaces
+
+* Does the PR change a public interface (e.g. API endpoints, clients)? If so:
+    * Do the interface changes match the spec?
+    * Are the names clear and obvious?
+    * Is there duplicate functionality?
+    * Will the change "box us in" for future functionality?
+    * Are there sensible defaults?
+* If the change touches the inference API:
+    * Does the PR also add the change to the OpenAI compatibility layer, the embedded Python client, and NAPI?
+    * Does the PR add tests for the OpenAI client changes in each language we test (Python, Node, Go)?
+* If the change touches any CRUD-style API:
+    * Does this match the patterns of the `v1/` APIs as much as possible?
+
+### Configuration
+
+Does the PR touch the configuration format? If so:
+* If we are breaking an existing interface, are we deprecating gracefully?
+* Are the names clear and obvious?
+* Is there duplicate functionality?
+* Will the change "box us in" for future functionality?
+* Are there sensible defaults?
+* Is any new config in the most obvious section?
+
+### Data model
+
+Are there changes to anything being written to Postgres or ClickHouse? If so:
+* Are there tests covering that what is written can be read as expected?
+* Are we taking care not to invalidate historical data? Is this tested?
+* Is the new format for writing extensible? (e.g. always tag enums)
+
+### Error Handling
+
+* If an error is because of bad input: is it obvious what was bad and how one might fix it?
+* If an error could be fixed by enabling a setting: add instructions on how to resolve it.
+
+### UI
+
+* Are there any large React components? Please try and factor those.
+* Are there any new fetches in components? Consider a hook.
+* Be extra careful with `useEffect`.
+
+### Testing
+
+#### Coverage
+
+Rust:
+* New gateway features that affect calls to providers should exercise the providers.
+  The ideal way to do this would be to ask the LLM something that it needs to use the functionality to answer.
+  Sometimes this is impossible, but think about if it could be done.
+* Any function with nontrivial logic should be unit-tested.
+
+UI:
+* New UI features should come with Playwright tests.
+* Nontrivial components should have Storybook stories.
+* Any tricky code should be unit tested with Vitest.
+
+#### Reliability
+
+* Try and factor out repeated test code into helpers.
+* Are the tests well-isolated? Will writing another test break them?
+* Are the tests going to require a lot of compute? disk? memory?
+* As much as possible: can we cache flaky network responses?
+
+#### Consolidation
+
+* Can tests be combined into units that preserve coverage but reduce the overall volume of code? Helpful for saving context and test runtime.
+
+### Performance
+* Is there anything that could be expensive added to the latency-critical path for inference?
```

Comment:
nit: "Is the PR adding anything expensive to inference requests, possibly on the critical path?" - use active voice, and "critical path" is a specific thing - https://en.wikipedia.org/wiki/Critical_path_method

Also unrelated but we should understand how to profile rust code and PyO3 and requests in the axum server.